tests/goal-workflow.test.sh

```
@@ -0,0 +1,567 @@
+# N — the nesting budget
+# ---------------------------------------------------------------------------
+echo "== N: the one-level nesting budget =="
+
+N_DRIVER="$(grep -cE '(^|[^A-Za-z0-9_])workflow\(\{' "$WORKFLOW" || true)"
+if [ "$N_DRIVER" = "1" ]; then
+  pass "N1 the driver has exactly ONE nested workflow() call site (count=$N_DRIVER)"
+else
+  fail "N1 expected exactly 1 nested workflow() site in the driver, found $N_DRIVER — a per-issue nested call spends the single nesting level on the wrong thing and the fleet's own agents cannot run"
+fi
+grep -q 'skills/solve-fleet/workflow.js' "$WORKFLOW" \
+  && pass "N2 the nested call targets skills/solve-fleet/workflow.js" \
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)